File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxProfit=0
        left=0
        right=left+1
        while(right<len(prices)):
            if(prices[right]<prices[left]):
                left=right
            else:
                maxProfit=max(prices[right]-prices[left], maxProfit)
            right+=1
        return maxProfit

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.java
class Solution {
    public int characterReplacement(String s, int k) {
        int[] count = new int[26];
        int windowCount = 0;
        int maxLength = 0;
        int left = 0;
        for(int r=0; r<s.length(); r++){
            count[s.charAt(r)-'A']++;
            windowCount = Math.max(windowCount, count[s.charAt(r)-'A']);
            if((((r-left)+1)-windowCount)>k){
                count[s.charAt(left)-'A']--;
                left++;
            }
            maxLength=Math.max(maxLength, (r-left)+1);
        }
        return maxLength;
    }
}
